### PR TITLE
Add the capability to use RoleBindings instead of ClusterRoleBindings

### PR DESCRIPTION
**Pull request by nilushancosta**
### Description
This PR introduces a new key to the Helm values named `useRoleBindings` which when set to `true` creates Kubernetes RoleBindings instead of ClusterRoleBindings

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-k8s-operator/issues/831

### Check List
- [x] Commits are signed per the DCO using --signoff 
- [ ] Unittest added for the new/changed functionality and all unit tests are successful
- [x] Customer-visible features documented
- [ ] No linter warnings (`make lint`)

If CRDs are changed:
- [ ] CRD YAMLs updated (`make manifests`) and also copied into the helm chart
- [ ] Changes to CRDs documented

Please refer to the [PR guidelines](https://github.com/opensearch-project/opensearch-k8s-operator/blob/main/docs/developing.md#submitting-a-pr) before submitting this pull request.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nilushancosta** [comment]
@swoehrl-mw would you be able to have a look at this PR?

**Comment by nilushancosta** [comment]
> I think it makes more sense to have the ClusterRoleBinding and the RoleBinding in the same file (without the `-crb` in the filename to make it more generic) and use an if-else-end construct.

@swoehrl-mw I changed it as requested

File: charts/opensearch-operator/templates/opensearch-operator-manager-rolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-manager-role
+subjects:
+- kind: ServiceAccount
+  name: {{ include "opensearch-operator.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+{{- else }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-manager-cluster-rolebinding
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/templates/opensearch-operator-manager-rolebinding.yaml:18]:
```suggestion
  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-manager-rolebinding
```

Name should stay the same as it is now, IMO it is also not a problem if rolebinding and clusterrolebinding are named the same

File: charts/opensearch-operator/templates/opensearch-operator-proxy-rolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-proxy-role
+subjects:
+- kind: ServiceAccount
+  name: {{ include "opensearch-operator.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+{{- else }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-proxy-cluster-rolebinding
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/templates/opensearch-operator-proxy-rolebinding.yaml:18]:
```suggestion
  name: {{ include "opensearch-operator.fullname" . }}-{{ .Release.Namespace }}-proxy-rolebinding
```

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]

**Review by swoehrl-mw — APPROVED** [review]

**Pull request merged**